src/Nest/Indices/MappingManagement/GetMapping/GetMappingResponse.cs

```diff
@@ -44,6 +44,13 @@ public class GetMappingResponse : DictionaryResponseBase<IndexName, IndexMapping
 		[JsonIgnore]
 		public IReadOnlyDictionary<IndexName, IndexMappings> Mappings => Indices;
 
+		[Obsolete("Use GetMappingFor explicitly instead this is a leaky abstraction that returns the mapping of the first index's first type on the response")]
+		public TypeMapping Mapping => this.Indices.FirstOrDefault().Value?.Mappings?.FirstOrDefault().Value;
+
+		public TypeMapping GetMappingFor<T>() => this.Indices[typeof(T)]?[typeof(T)];
+		public TypeMapping GetMappingFor(string index, string type) => this.Indices[index]?[type];
+		public TypeMapping GetMappingFor(string index) => this.Indices[index]?.Mappings?.FirstOrDefault().Value;
```

Comment:
Nice little helper, but I wonder if this should be SingleOrDefault() so it throws if you have more than 1 mapping (to let you know you might get a random mapping back. Also this needs a null check after the FirstOrDefault()?.Value

Reply:
Multiple mappings is to be expected in this case (and the initial reason for removing the `.Mapping` helper property.